python/pacemaker/_cts/tests/cibsecret.py

```diff
@@ -0,0 +1,176 @@
+
+        self._cm.add_dummy_rsc(node, self._rid)
+
+        with Timer(self._logger, self.name, "addDummy"):
+            watch.look_for_all()
+
+        if watch.unmatched:
+            self.debug("Failed to find patterns when adding dummy resource")
+            return repr(watch.unmatched)
+
+        return ""
+
+    def _check_cib_value(self, node, expected):
+        (rc, lines) = self._rsh(node, f"crm_resource -r {self._rid} -g {self._secret}",
+                                verbose=1)
+        s = " ".join(lines).strip()
+
+        if rc != 0 or s != expected:
+            return self.failure(f"Secret set to '{s}', not '{expected}'")
+
+        # This is self.success, except without incrementing the success counter
+        return True
+
+    def _test_check(self, node):
+        (rc, _) = self._rsh(node, f"cibsecret check {self._rid} {self._secret}",
+                            verbose=1)
+        if rc != 0:
+            return self.failure("Failed to check secret")
+
+        # This is self.success, except without incrementing the success counter
+        return True
+
+    def _test_delete(self, node):
+        (rc, _) = self._rsh(node, f"cibsecret delete {self._rid} {self._secret}",
+                            verbose=2)
+        if rc != 0:
+            return self.failure("Failed to delete secret")
+
+        # This is self.success, except without incrementing the success counter
+        return True
+
+    def _test_get(self, node, expected):
+        (rc, lines) = self._rsh(node, f"cibsecret get {self._rid} {self._secret}",
+                                verbose=1)
+        s = " ".join(lines).strip()
+
+        if rc != 0 or s != expected:
+            return self.failure(f"Secret set to '{s}', not '{expected}'")
+
+        # This is self.success, except without incrementing the success counter
+        return True
+
+    def _test_set(self, node):
+        (rc, _) = self._rsh(node, f"cibsecret set {self._rid} {self._secret} {self._secret_val}",
+                            verbose=2)
+        if rc != 0:
+            return self.failure("Failed to set secret")
+
+        # This is self.success, except without incrementing the success counter
+        return True
+
+    def _test_stash(self, node):
+        (rc, _) = self._rsh(node, f"cibsecret stash {self._rid} {self._secret}",
+                            verbose=2)
+        if rc != 0:
+            return self.failure(f"Failed to stash secret {self._secret}")
+
+        # This is self.success, except without incrementing the success counter
+        return True
+
+    def _test_unstash(self, node):
+        (rc, _) = self._rsh(node, f"cibsecret unstash {self._rid} {self._secret}",
+                            verbose=2)
+        if rc != 0:
+            return self.failure(f"Failed to unstash secret {self._secret}")
+
+        # This is self.success, except without incrementing the success counter
+        return True
+
+    def __call__(self, node):
```

Comment:
Could also test some error cases if desired. For example, trying to stash a value that's already secret, and then looking for an expected return code (passed as an argument)